### PR TITLE
Mongo setup

### PR DESCRIPTION
**Pull request by shrikantr-webonise**

### CODE REVIEW

File: app/lib/constants.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  fields: {
+    select_all: {
+      name: true,
+      description: true,
+      completed: true,
+      updatedAt: true,
+      deleted: true
+    },
+  },
+  response_message: {
```
Comment (eliza-abraham) [inline app/lib/constants.js]: We can move these to en.yml

File: app/lib/db.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      mongoose.connection.on('open', function () {
+        logger.info('Connected to database.');
+        let names = [];
+        return mongoose.connection.db.listCollections().toArray(function (err, collections) {
+          collections.forEach((collection) => {
+            names.push(collection.name);
+          });
+
+          if (names.indexOf('todos') === -1) {
```
Comment (eliza-abraham) [inline app/lib/db.js]: we can do this dynamically. We shouldn't have to statically define this for each new model. 
We can pull a list of all the model names from the model folder, compare this names against existing collections and then send warning messages for each model.

File: app/controllers/todo.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+/*
+ * Todo Controller to handle express requests.
+ * Parameters common for all methods.
+ * @param {object} request - Request object of express.
+ * @param {object} response - Response object of express.
+ * @return {callback} - A callback to response for HTTP.
+*/
+class TodoController {
+
+  findTodos (request, response) {
```
Comment (eliza-abraham) [inline app/controllers/todo.js]: Remove the redundancy in naming. Todo is implicit from the controller name

File: app/lib/constants.js (hunk 1 above)
Comment (shrikantr-webonise) [reply app/lib/constants.js]: Will do as an improvement later.

**Pull request merged**